src/qml/qmlautoreload.cpp

```diff
@@ -18,25 +18,40 @@ QmlAutoReload::QmlAutoReload() {
 }
 
 QUrl QmlAutoReload::intercept(const QUrl& url, QQmlAbstractUrlInterceptor::DataType type) {
-    if (!url.isLocalFile() || !QFileInfo(url.toLocalFile()).isFile()) {
+    const auto path = url.toLocalFile();
+
+    if (!url.isLocalFile() || !QFileInfo(path).isFile()) {
+        return url;
+    }
+
+    // Do not add path if it's already watched. See note in the
+    // `QFileSystemWatcher::fileChanged()` documentation.
+    if (m_fileWatcher.files().contains(path)) {
```

Comment:
This is redundant as `[addPath](https://doc.qt.io/qt-6/qfilesystemwatcher.html#addPaths)` we already perform this check.

> Paths are not added if they not exist, or if they are already being monitored by the file system watcher

Reply:
Alright, removed.